### ISSUE TITLE
NavLink does not update Route parameters in code

### ISSUE THREAD

**Issue by JoP96** [aspnet/AspNetCore#9931]
My .razor file looks as follows:
```
@page "/example"
@page "/example/{id}"

<h1>@ID</h1>
<NavLink href="/example/something">Something</NavLink>
<NavLink href="/example/">Nothing</NavLink>
<button onclick="@Show">Show</button>

@functions {
    [Parameter]
    public string ID { get; set; } = "Nothing";

    public void Show()
    {
        Console.WriteLine(ID);
    }
}
```

When I navigate to /example, the h1 tag contains "Nothing", as expected. If I now click on the NavLink "Something", the h1 tag contains "Something", also as expected.
But if I then click on the NavLink "Nothing", the URL in browser changes the right way, but the h1 tag still contains "Something".
If I check the value of by clicking the button, the console shows me "Something".

Is this a bug or a feature? I would have expected the h1 tag to contain an empty string.

If this is a feature, I would recommand to change that, because this makes it impossible to navigate from /example/something back to /example.

![image](https://user-images.githubusercontent.com/16904516/57100207-9fdcb980-6d1e-11e9-91d6-ad2321ceca85.png)

---

My setup:
- VS 2019
- ASP.NET Core 3.0
- Microsoft.AspNetCore.Blazor from NuGet (Version 3.0.0-preview4-19216-03)
- Default Blazor (ASP.NET Core hosted) Project

**Comment by Andrzej-W** [comment]
This bug is tracked here #7419. You can close because it is a duplicate.

**Comment by Andrzej-W** [comment]
https://github.com/aspnet/AspNetCore/issues/7419#issuecomment-462096315
> This is very old bug and it was tracked somewhere. It was fixed but then reverted. See this #6864. Hopefully both problems will be solved in preview 3.

https://github.com/aspnet/AspNetCore/issues/7419#issuecomment-465098832
> The change where we originally fixed this was #4797, but then we reverted it in #6931.

We have to wait for the fix.

**Comment by JoP96** [comment]
Ok, thanks!
